src/hooks/room/useRoomCallStatus.ts

```diff
@@ -0,0 +1,146 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { Room, RoomStateEvent } from "matrix-js-sdk/src/matrix";
+import { useCallback, useEffect, useMemo, useState } from "react";
+
+import { useFeatureEnabled } from "../useSettings";
+import SdkConfig from "../../SdkConfig";
+import { useEventEmitterState, useTypedEventEmitterState } from "../useEventEmitter";
+import LegacyCallHandler, { LegacyCallHandlerEvent } from "../../LegacyCallHandler";
+import { useWidgets } from "../../components/views/right_panel/RoomSummaryCard";
+import { WidgetType } from "../../widgets/WidgetType";
+import { useCall } from "../useCall";
+import { _t } from "../../languageHandler";
+import { useRoomMembers } from "../useRoomMembers";
+import { ElementCall } from "../../models/Call";
+
+type CallType = "element_call" | "jitsi_or_element_call" | "legacy_or_jitsi";
+
+/**
+ * Reports the call capabilities for the current room
+ * @param room the room to track
+ * @returns the call status for a room
+ */
+export const useRoomCallStatus = (
+    room: Room,
+): {
+    voiceCallDisabledReason: string | null;
+    voiceCallType: CallType;
+    videoCallDisabledReason: string | null;
+    videoCallType: CallType;
+} => {
+    const [voiceCallDisabledReason, setVoiceCallDisabledReason] = useState<string | null>(null);
+    const [videoCallDisabledReason, setVideoCallDisabledReason] = useState<string | null>(null);
+    const [voiceCallType, setVoiceCallType] = useState<CallType>("jitsi_or_element_call");
+    const [videoCallType, setVideoCallType] = useState<CallType>("jitsi_or_element_call");
+
+    const groupCallsEnabled = useFeatureEnabled("feature_group_calls");
+    const useElementCallExclusively = useMemo(() => {
+        return SdkConfig.get("element_call").use_exclusively;
+    }, []);
+
+    const hasLegacyCall = useEventEmitterState(
+        LegacyCallHandler.instance,
+        LegacyCallHandlerEvent.CallsChanged,
+        useCallback(() => LegacyCallHandler.instance.getCallForRoom(room.roomId) !== null, [room]),
+    );
+
+    const widgets = useWidgets(room);
+    const hasJitsiWidget = useMemo(() => widgets.some((widget) => WidgetType.JITSI.matches(widget.type)), [widgets]);
+
+    const hasGroupCall = useCall(room.roomId) !== null;
+
+    const functionalMembers = useRoomMembers(room);
+
+    const [mayEditWidgets, mayCreateElementCalls] = useTypedEventEmitterState(
+        room,
+        RoomStateEvent.Update,
+        useCallback(
+            () => [
+                room.currentState.mayClientSendStateEvent("im.vector.modular.widgets", room.client),
+                room.currentState.mayClientSendStateEvent(ElementCall.CALL_EVENT_TYPE.name, room.client),
+            ],
+            [room],
+        ),
+    );
+
+    useEffect(() => {
+        if (groupCallsEnabled) {
+            if (useElementCallExclusively) {
+                if (hasGroupCall) {
+                    // done
```

Comment:
What does „done“ mean here? If it is done done, I would expect a `return`. Can you please update or remove the comments?

Reply:
Sorry, that's a leftover from my testing. I was tracking the conditions i covered with unit tests.

The logic is pretty much copied over from https://github.com/matrix-org/matrix-react-sdk/blob/bba980cd8a55e4c9bdf18eaec3ab29174b6a73da/src/components/views/rooms/LegacyRoomHeader.tsx#L291